synth_data_module/formatting_helpers.py

```diff
@@ -0,0 +1,151 @@
+    procedure_list = []
+    for i in range(1, 26):
+        if i == 1:
+            code = {'fieldid': 'proc_p', 'name': f'Principal Procedure Code', 'length': 7, 'justification': 'left'}
+            procedure_list.append(code)
+            date = {'fieldid': 'proc_pdt', 'name': f'Principal Procedure Date', 'length': 8, 'justification': 'left'}
+            procedure_list.append(date)
+            date = {'fieldid': 'proc_pdy', 'name': f'Principal Procedure Days', 'length': 8, 'justification': 'left'}
+            procedure_list.append(date)
+        else:
+            code = {'fieldid': f'oproc{i}', 'name': f'Procedure Code {i}', 'length': 7, 'justification': 'left'}
+            procedure_list.append(code)
+            date = {'fieldid': f'procdt{i}', 'name': f'Procedure Date {i}', 'length': 8, 'justification': 'left'}
+            procedure_list.append(date)
+            date = {'fieldid': f'procdy{i}', 'name': f'Procedure Days {i}', 'length': 8, 'justification': 'left'}
+            procedure_list.append(date)
+
+    return procedure_list
+
+
+def get_diagnosis_list(length):
+    diagnosis_list = []
+    for i in range(2, 26):
+        diagnosis = {'fieldid': f'odiag{i}', 'name': f'Diagnosis {i}', 'length': length, 'justification': 'left'}
+        diagnosis_list.append(diagnosis)
+        present_on_admission = {'fieldid': f'opoa{i}', 'name': f'Present on Admission {i}', 'length': 1, 'justification': 'left'}
+        diagnosis_list.append(present_on_admission)
+
+    return diagnosis_list
+
+
+def get_morbidity_list():
+    morbidity_list = []
+    for i in range(1, 13):
+        morbidity = {'fieldid': f'ecm{i}', 'name': f'External Causes of Morbidity {i}', 'length': 8, 'justification': 'left'}
+        morbidity_list.append(morbidity)
+        present_on_admission = {'fieldid': f'epoa{i}', 'name': f'POA - External Causes of Morbidity {i}', 'length': 1,
+                                'justification': 'left'}
+        morbidity_list.append(present_on_admission)
+
+    return morbidity_list
+
+
+def modify_row(row, df_fields, new_fields):
+    first = row[df_fields[0]]
+    second = row[df_fields[1]]
+    if True:  # 'Procedure Codes' in df_fields:
+        if isinstance(first, tuple) and isinstance(second, tuple):
+            for code, date, i in zip(first, second, range(1, 26)):
+                if i == 1:
+                    if new_fields[0] == 'Diagnosis':
+                        row[f'Principal {new_fields[0]}'] = code
+                        row[f'{new_fields[1]} for Principal {new_fields[0]}'] = date
+                    else:
+                        row[f'Principal {new_fields[0]}'] = code
+                        row[f'Principal {new_fields[1]}'] = date
+                else:
+                    row[f'{new_fields[0]} {i}'] = code
+                    row[f'{new_fields[1]} {i}'] = date
+    return row
+
+# HCAI has specific criteria for age reporting
+def calculate_age(x, y, form="agedays"):
```

Comment:
Comments at the function or class level should use[ triple-quote style comments](https://realpython.com/python-comments-guide/#when-writing-code-for-others)  aka docstrings. Most editors understand how to display these when you hover over a function name, and they show up when you run `help(func)` in the interpreter, `ipython`, or  `jupyter`.